cli/flags.go

```diff
@@ -300,8 +300,8 @@ func Flag[T any](f *FlagSection, i *Var[T]) {
 		}
 	}
 
-	// Set a default value.
-	*i.Target = initial
+	// Use the setter to set the default value.
+	setter(i.Target, initial)
```

Comment:
This was a bug. If the setter is used to set value, the original logic won't trigger it to set the default value.

Reply:
Sent a separate PR